New Code Format with pretty expansion and JSDoc

akanellop opened the pull request:
…s and css files

akanellop committed:
In package.json changed lint and format scripts so they include all js and css files

akanellop committed:
In package.json more specification given about lint and format script

akanellop committed:
added JSDoc on dependecies
started JSDoc for documentation
add doc directory to .gitignore

parisk commented on package.json:
```diff
@@ -15,10 +15,13 @@
   },
-  "dependencies": {}
+  "dependencies": {
+    "jsdoc": "^3.5.5"
```
Move `jsdoc` in `devDependencies` please. Explaining; env-spec-js does not depend on jsdoc itself to operate, it is a tool used by the developers of env-spec-js (us).

parisk commented on package.json:
```diff
@@ -15,10 +15,13 @@
     "watch": "watchify src/main.js --verbose --standalone envSpec -o dist/envspec.js",
-    "start": "browser-sync start --server demo --server dist --watch --files 'dist/*' 'demo/*'  --port $PORT"
+    "start": "browser-sync start --server demo --server dist --watch --files 'dist/*' 'demo/*'  --port $PORT",
+    "doc": "jsdoc src/main.js -d docs"
```
Rename this script to `docs` please.

parisk commented on src/main.js:
```diff
@@ -94,13 +115,21 @@ const parseVarFromType = envSpecAsArray => {
 
-//function that creates the render label for the output outputHTML
+/** @function renderLabelForEntry
+ * @desc Creates the render label for the {@link outputHTML} function.
```
> Creates the render label for the

We should prefer: *Creates the render for the*

"Render" is a verb we use to express the transformation of code into something visible by a human (e.g. HTML code is being rendered into an actual button).

akanellop committed:
altered package.json : changed script's name for jsdoc
changed dependecies to devDependecies
altered main.js from src directory : more specific description for jsdoc

akanellop committed:
readded jsdoc to devDependecies

akanellop committed:
cleared the directory from unescessary files
